crates/lang-node/src/yarn/mod.rs

```diff
@@ -0,0 +1,119 @@
+use cached::proc_macro::cached;
+use moon_error::MoonError;
+use moon_lang::LockfileDependencyVersions;
+use std::collections::HashMap;
+use std::fs;
+use std::path::PathBuf;
+use yarn_lock_parser::{parse_str, Entry};
+
+#[cached(result)]
+pub fn load_lockfile_dependencies(path: PathBuf) -> Result<LockfileDependencyVersions, MoonError> {
+    let mut deps: LockfileDependencyVersions = HashMap::new();
+
+    let yarn_lock_text = fs::read_to_string(&path)?;
+    let entries: Vec<Entry> = parse_str(&yarn_lock_text)
+        .map_err(|_| MoonError::Generic("Failed to parse lockfile".to_owned()))?;
+    for entry in entries {
+        if entry.integrity.is_empty() {
+            // all workspace dependencies have empty integrities, so we will skip them
```

Comment:
Does this mean `workspace:`?

Reply:
Yes. Basically the `workspaces` key of `package.json`.